test/fakeintake/client/client.go

```diff
@@ -512,6 +545,124 @@ func (c *Client) GetProcessDiscoveries() ([]*aggregator.ProcessDiscoveryPayload,
+// all received container image names
+func (c *Client) GetContainerImageNames() ([]string, error) {
+	if err := c.getContainerImages(); err != nil {
+		return nil, err
+	}
+	return c.containerImageAggregator.GetNames(), nil
+}
+
+// FilterContainerImages fetches fakeintake on `/api/v2/contimage` endpoint and returns
+// container images matching `name` and any [MatchOpt](#MatchOpt) options
+func (c *Client) FilterContainerImages(name string, options ...MatchOpt[*aggregator.ContainerImagePayload]) ([]*aggregator.ContainerImagePayload, error) {
+	images, err := c.getContainerImage(name)
+	if err != nil {
+		return nil, err
+	}
+	// apply filters one after the other
+	filteredImages := []*aggregator.ContainerImagePayload{}
+	for _, image := range images {
+		matchCount := 0
+		for _, matchOpt := range options {
+			isMatch, err := matchOpt(image)
+			if err != nil {
+				return nil, err
+			}
+			if !isMatch {
+				break
+			}
+			matchCount++
+		}
+		if matchCount == len(options) {
+			filteredImages = append(filteredImages, image)
+		}
+	}
+	return filteredImages, nil
+}
+
+// GetContainerLifecycleEvents fetches fakeintake on `/api/v2/contlcycle` endpoint and returns
+// all received container lifecycle payloads
+func (c *Client) GetContainerLifecycleEvents() ([]*aggregator.ContainerLifecyclePayload, error) {
+	if err := c.getContainerLifecycleEvents(); err != nil {
+		return nil, err
+	}
+
+	var events []*aggregator.ContainerLifecyclePayload
+	for _, name := range c.containerLifecycleAggregator.GetNames() {
+		events = append(events, c.containerLifecycleAggregator.GetPayloadsByName(name)...)
+	}
+
+	return events, nil
+}
+
+func (c *Client) getSBOM(id string) ([]*aggregator.SBOMPayload, error) {
+	if err := c.getSBOMs(); err != nil {
+		return nil, err
+	}
+	return c.sbomAggregator.GetPayloadsByName(id), nil
+}
+
+// GetSBOMIDs fetches fakeintake on `/api/v2/sbom` endpoint and returns
+// all received SBOM IDs
+func (c *Client) GetSBOMIDs() ([]string, error) {
+	if err := c.getSBOMs(); err != nil {
+		return nil, err
+	}
+	return c.sbomAggregator.GetNames(), nil
+}
+
+// FilterSBOMs fetches fakeintake on `/api/v2/sbom` endpoint and returns
+// SBOMs matching `id` and any [MatchOpt](#MatchOpt) options
+func (c *Client) FilterSBOMs(id string, options ...MatchOpt[*aggregator.SBOMPayload]) ([]*aggregator.SBOMPayload, error) {
+	sboms, err := c.getSBOM(id)
+	if err != nil {
+		return nil, err
+	}
+	// apply filters one after the other
+	filteredSBOMs := []*aggregator.SBOMPayload{}
+	for _, sbom := range sboms {
+		matchCount := 0
+		for _, matchOpt := range options {
+			isMatch, err := matchOpt(sbom)
+			if err != nil {
+				return nil, err
+			}
+			if !isMatch {
+				break
+			}
+			matchCount++
+		}
+		if matchCount == len(options) {
+			filteredSBOMs = append(filteredSBOMs, sbom)
+		}
+	}
+	return filteredSBOMs, nil
+}
+
+// // GetSBOMs fetches fakeintake on `/api/v2/sbom` endpoint and returns
```

Comment:
Perhaps we want to remove these comments

Reply:
Aren’t they required by linters ?